onnxruntime/core/providers/dml/DmlExecutionProvider/src/DmlGraphFusionTransformer.cpp

```diff
@@ -137,38 +137,39 @@ namespace Dml
                 //  backed resource, avoiding an extra set of GPU resources in memory.
                 std::unordered_map<std::string, std::pair<const ONNX_NAMESPACE::TensorProto*, bool>> isInitializerTransferable;
 
-                if (partition->IsDmlGraphPartition())
+            
+            if (partition->IsDmlGraphPartition())
+            {
+                // populate transferredInitializerMap
+                for (const auto& input : partition->GetInputs())
                 {
-                    // populate isInitializerTransferable
-                    for (const auto& input : partition->GetInputs())
+                    const onnx::TensorProto* tensor = nullptr;
+                    if (graph.GetInitializedTensor(input, tensor))
                     {
-                        const onnx::TensorProto* tensor = nullptr;
-                        if (graph.GetInitializedTensor(input, tensor))
+                        // It's only safe to transfer tensors which are used by this partition alone.
+                        auto iter = initializerPartitionMap.find(tensor);
+                        assert(iter != initializerPartitionMap.end());
```

Comment:
```suggestion
                        ORT_ENFORCE(iter != initializerPartitionMap.end());
```
Just in case we have a logic error in the future, the error will be exposed, rather than silently reading past the end of the array.